posthog/models/tagged_item.py

```diff
@@ -2,23 +2,42 @@
 
 from django.core.exceptions import ValidationError
 from django.db import models
-from django.db.models import Q
+from django.db.models import Q, UniqueConstraint
 
 from posthog.models.utils import UUIDModel
 
 RELATED_OBJECTS = ("dashboard", "insight", "event_definition", "property_definition", "action")
 
 
+# Checks that exactly one object field is populated
 def build_check():
     built_check_list: List[Union[Q, Q]] = []
-    # Only one object field can be populated
     for o_field in RELATED_OBJECTS:
         built_check_list.append(
             Q(*[(f"{_o_field}__isnull", _o_field != o_field) for _o_field in RELATED_OBJECTS], _connector="AND")
         )
     return Q(*built_check_list, _connector="OR")
 
 
+# Enforces uniqueness on tag_{object_field}. All permutations of null columns must be explicit as Postgres ignores
+# uniqueness across null columns.
+def build_uniqueness_constraint():
+    built_check_list: List[UniqueConstraint] = [
+        UniqueConstraint(fields=("tag",) + RELATED_OBJECTS, name=f"unique_tagged_item"),
```

Comment:
What does this constraint do? I understand the others, but not sure I get this one.

Super nit: the name string doesn't have to be a f-string

Reply:
Ah good catch, this is a case that the exactly one related object check already accounts for.